plugins/main/public/components/overview/overview.tsx

```diff
@@ -0,0 +1,223 @@
+      return Promise.reject(error);
+    }
+  };
+
+  function switchTab(newTab: any, force: any) {
+    try {
+      if (tabActive === newTab && !force) {
+        return;
+      }
+      setTabActive(newTab);
+      setTabViewActive($location.search().tabView || 'panels');
+    } catch (error) {
+      const options = {
+        context: `${Overview.name}.switchTab`,
+        level: UI_LOGGER_LEVELS.ERROR,
+        severity: UI_ERROR_SEVERITIES.BUSINESS,
+        error: {
+          error: error,
+          message: error.message || error,
+          title: error.name || error,
+        },
+      };
+
+      getErrorOrchestrator().handleError(options);
+    }
+  }
+
+  const switchSubTab = (subTab: string) => {
+    try {
+      if (tabViewActive !== subTab) {
+        setTabViewActive(subTab);
+      }
+    } catch (error) {
+      const options = {
+        context: `${Overview.name}.switchSubtab`,
+        level: UI_LOGGER_LEVELS.ERROR,
+        severity: UI_ERROR_SEVERITIES.BUSINESS,
+        error: {
+          error: error,
+          message: error.message || error,
+          title: error.name || error,
+        },
+      };
+
+      getErrorOrchestrator().handleError(options);
+    }
+  };
+  // TODO: Create a service or add it in the dataSource the management of the pinned agent
+  const updateSelectedAgents = (agentList: Array<any>) => {
+    try {
+      if (agentList.length > 0) {
+        if ($location.search().agentId !== agentList[0]) {
+          $location.search('agentId', agentList[0]);
+          $route.reload();
+        }
+      } else {
+        setTimeout(() => {
+          if ($location.search()['agentId']) {
+            $location.search('agentId', null);
+            $route.reload();
+          }
+        }, 1);
+      }
+    } catch (error) {
+      const options = {
+        context: `${Overview.name}.updateSelectedAgents`,
+        level: UI_LOGGER_LEVELS.ERROR,
+        severity: UI_ERROR_SEVERITIES.BUSINESS,
+        error: {
+          error: error,
+          message: error.message || error,
+          title: error.name || error,
+        },
+      };
+
+      getErrorOrchestrator().handleError(options);
+    }
+  };
+
+  return (
+    <>
+      {tabActive && tabActive !== 'welcome' && (
+        <>
+          <MainModule
+            section={tabActive}
+            agentsSelectionProps={{
+              tab: tabActive,
+              subtab: tabViewActive,
+              setAgent: (agentList: Array<any>) =>
+                updateSelectedAgents(agentList),
+            }}
+            switchSubTab={subTab => switchSubTab(subTab)}
+          />
+          <WzCurrentOverviewSectionWrapper
+            switchTab={(tab, force) => switchTab(tab, force)}
+            currentTab={tabActive}
+          />
+        </>
+      )}
+      {tabActive === 'welcome' && (
+        <>
+          <Stats {...agentsCounts} />
+          <OverviewWelcome {...agentsCounts} />
+        </>
+      )}
+    </>
+  );
+};
```

Comment:
I understand that everything related to TabVisualizations, VisFactoryHandler and RawVisualizations would no longer be necessary. No?